service/storage_proxy.cc

```diff
@@ -809,7 +813,11 @@ class storage_proxy::remote {
                 return p->query_result_local(erm, std::move(s), cmd, std::move(pr2.first), opts, trace_state_ptr, timeout, rate_limit_info);
             } else if constexpr (verb == read_verb::read_mutation_data) {
                 p->get_stats().replica_mutation_data_reads++;
-                return p->query_mutations_locally(std::move(s), std::move(cmd), pr2, timeout, trace_state_ptr);
+                return p->query_mutations_locally(std::move(s), std::move(cmd), pr2, timeout, trace_state_ptr).then([format_reverse_required](auto r)
+                    {
+                        auto&& [result, hit_rate] = r;
+                        return rpc::tuple{format_reverse_required ? make_foreign(reversed(result.release())) : std::move(result), hit_rate};
+                    });
```

Comment:
To avoid an allocation in the common case (upgraded cluster), apply the continuation conditionally.

(e.g. `f = query_mutations_locally(); if (reverse required) { f = f.then(...); }`)